src/commands/lambda/__tests__/flare.test.ts

```diff
@@ -1,21 +1,444 @@
+      const cli = makeCli()
+      const context = createMockContext()
+      const code = await cli.run(
+        ['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-e', 'test@test.com'],
+        context as any
+      )
+      expect(code).toBe(1)
+      const output = context.stderr.toString()
+      expect(output).toMatchSnapshot()
+    })
+
+    it('prints error when no email specified', async () => {
+      const cli = makeCli()
+      const context = createMockContext()
+      const code = await cli.run(['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-c', '123'], context as any)
+      expect(code).toBe(1)
+      const output = context.stderr.toString()
+      expect(output).toMatchSnapshot()
+    })
+
+    it('runs successfully with all required options specified', async () => {
+      const cli = makeCli()
+      const context = createMockContext()
+      const code = await cli.run(VALID_INPUT, context as any)
+      expect(code).toBe(0)
+      const output = context.stdout.toString()
+      expect(output).toMatchSnapshot()
+    })
+  })
+
+  describe('writeFile', () => {
+    const MOCK_DATA = 'mock data'
+    const instance = new LambdaFlareCommand()
+    ;(fs.existsSync as jest.Mock).mockReturnValue(false)
+
+    it('successfully writes data to a file with no error', async () => {
+      await writeFile(MOCK_FOLDER_PATH, MOCK_FILE_PATH, MOCK_DATA, instance.context)
+
+      expect(fs.existsSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
+      expect(fs.mkdirSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
+      expect(fs.writeFileSync).toHaveBeenCalledWith(MOCK_FILE_PATH, MOCK_DATA)
+    })
+
+    it('throws error when unable to create folder', async () => {
+      ;(fs.mkdirSync as jest.Mock).mockImplementation(() => {
+        throw new Error('MOCK ERROR: Unable to create folder')
+      })
+
+      await expect(writeFile(MOCK_FOLDER_PATH, MOCK_FILE_PATH, MOCK_DATA, instance.context)).rejects.toMatchSnapshot()
+
+      expect(fs.existsSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
+      expect(fs.mkdirSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
+      ;(fs.mkdirSync as jest.Mock).mockRestore()
+    })
+
+    it('throws error when unable to write data to a file', async () => {
+      ;(fs.writeFileSync as jest.Mock).mockImplementation(() => {
+        throw new Error('MOCK ERROR: Unable to write file')
+      })
+
+      await expect(writeFile(MOCK_FOLDER_PATH, MOCK_FILE_PATH, MOCK_DATA, instance.context)).rejects.toMatchSnapshot()
+
+      expect(fs.existsSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
+      expect(fs.mkdirSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
+      expect(fs.writeFileSync).toHaveBeenCalledWith(MOCK_FILE_PATH, MOCK_DATA)
+    })
+  })
+
+  describe('zipContents', () => {
+    beforeEach(() => {
+      fs.writeFileSync = jest.fn().mockImplementation(() => {})
+      fs.readFileSync = jest.fn().mockResolvedValue(JSON.stringify(MOCK_CONFIG, undefined, 2))
```

Comment:
Aren't these two already defined at the top of the file?

Reply:
Yes, but some tests in the describe block edit the implementation of these two. For example:
https://github.com/DataDog/datadog-ci/blob/4e84a9221001b2deb0c2bfe502c9b7b377e65fb2/src/commands/lambda/__tests__/flare.test.ts#L267-L269

Reply:
Let's leave this work for another PR. Let's log a ticket for this, since other people jumping on the project might have a hard time understanding all of what's going on with the mocks.